### DISCUSSION TITLE
Spawning Creatures in Campaign via Item or otherwise causing a stacking issue that requires a server reset to clear

### DISCUSSION THREAD

**Discussion by Draconis13** [FakeFishGames/Barotrauma#14367] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

Using Violet's mod Baroverhaul - Classic a Thalamus was loaded with the custom Dementonite coilgun ammo which spawns 1 short lived swarm feeder on hit, effectively useless against players as there are human team, however the issue was after it fired three fifths of the box before being disable lag got noticeably bad on the server. Once we switched levels from mission to station the problem persisted, I went back to lobby and returned to game and still same high numbers in Showperf in console. I fired the rest of the coil gun box and the lag shot up even higher and same issue persisted. I looked over the code and there is nothing going on other than a creature that spawns, dies of internal damage and is removed rapidly. 

This only happens in campaign and not in submarine editor and simply making a custom gun or item that on use spawns a completely neutral character without any sounds or special effects causes this. I think it might be something clogging up the AI controller and not being discarded?  

Fairly easy to repeat by making a SMG or other onuse item that spawns crawlers/swarm feeders which spawn in, die and are removed. I am thinking this might be affecting several other mods that spawn in characters outside of the event sets like trap items etc in barotraumatic. as well the base game egg item code but I would need to test it more.  Have not checked dedicated servers, only in game host server function. 

![image](https://github.com/user-attachments/assets/48f318bd-1671-4541-a94f-0ed9f1e46495)

![Barotrauma_mM8wKNbmSO](https://github.com/user-attachments/assets/8daa6183-85f7-4d5c-8e75-1e6a9a862035)

### Reproduction steps

Spawn multiple characters in using an item during campaign play, they die and are removed yet some portion of their AI think brain or other back end code persists and with each spawn the servers load increases. Changing levels or going to lobby does not fix issue and server must be fully restarted to clear problem. 

### Bug prevalence

Happens regularly

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report! I unfortunately could not find a way to reproduce the issue. In my tests, it seemed that the performance returned back to normal as soon as the creatures had been deleted:


https://github.com/user-attachments/assets/561f7cf7-5516-40d5-9c83-6e4b773f1e4b


https://github.com/user-attachments/assets/2034fbea-acd6-41e7-be3e-4eb828677fc7

**Comment by Draconis13** [comment]
Strange I don't know why this is happening on my game when I host multiplayer, we tested it in singleplayer and its like your above screenshots and everything returns to normal. I am also not running performance fix on the server in case of any desync/tick rate shenanigans. What really throws me for a loop is that going to lobby doesn't reset it but restarting the server does. meaning its not being cleared from memory for whatever reason.

The only difference on these swarmlings is they have a different sprite and are human/player team. I will try to test this more with custom monster spawners. I had previously noticed that spawning monsters in ruins from vents during campaign seemed to cause a bump in server load but with how ruins are especially in modded I didn't think too much of it.
